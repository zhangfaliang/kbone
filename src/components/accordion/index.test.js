import toJson from 'enzyme-to-json';
import { render, mount } from 'enzyme';
import { Accordion, MatchDetailPanel, Panel } from './index';

describe('Accordion', async () => {
  it('base', () => {
    expect.assertions(1);
    const wrapper = render(<Accordion accordion={true} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('base-accordion', () => {
    expect.assertions(1);
    const wrapper = render(<Accordion accordion={true} activeKeyId="" />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('basePanel', () => {
    expect.assertions(1);
    const wrapper = render(
      <Accordion accordion={true}>
        <Panel />
      </Accordion>
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('baseMatchDetailPanel', () => {
    expect.assertions(1);
    const wrapper = render(
      <Accordion>
        <MatchDetailPanel />
      </Accordion>
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('defaulthandlePanel', () => {
    expect.assertions(1);
    const wrapper = mount(
      <Accordion accordion={true}>
        <Panel />
      </Accordion>
    );
    const handlePanel = wrapper.prop('handlePanel');
    const click = jest.fn(handlePanel);
    wrapper
      .find('Accordion')
      .find('Panel')
      .children('div')
      .at(0)
      .children('div')
      .at(0)
      .simulate('click');
    expect(click()).toBe('handlePanel');
  });
  it('defaulthandlePanel-accordion', () => {
    expect.assertions(1);
    const wrapper = mount(
      <Accordion accordion={false}>
        <Panel />
      </Accordion>
    );
    const handlePanel = wrapper.prop('handlePanel');
    const click = jest.fn(handlePanel);
    wrapper
      .find('Accordion')
      .find('Panel')
      .children('div')
      .at(0)
      .children('div')
      .at(0)
      .simulate('click');
    expect(click()).toBe('handlePanel');
  });
  it('defaulthandleStickIcon', () => {
    expect.assertions(1);
    const wrapper = mount(
      <Accordion accordion={true}>
        <MatchDetailPanel />
      </Accordion>
    );
    const handleStickIcon = wrapper.prop('handleStickIcon');
    const click = jest.fn(handleStickIcon);
    wrapper
      .find('Accordion')
      .find('MatchDetailPanel')
      .children('div')
      .at(0)
      .children('div')
      .at(0)
      .children('div')
      .at(1)
      .children('span')
      .at(0)
      .simulate('click');
    expect(click()).toBe('handleStickIcon');
  });
  it('defaulthandleHelpIcon', () => {
    expect.assertions(1);
    const wrapper = mount(
      <Accordion accordion={true}>
        <MatchDetailPanel />
      </Accordion>
    );
    const handleHelpIcon = wrapper.prop('handleHelpIcon');
    const click = jest.fn(handleHelpIcon);
    wrapper
      .find('Accordion')
      .find('MatchDetailPanel')
      .children('div')
      .at(0)
      .children('div')
      .at(0)
      .children('div')
      .at(0)
      .children('span')
      .at(0)
      .simulate('click');
    expect(click()).toBe('handleHelpIcon');
  });
});

describe('MatchDetailPanel', async () => {
  it('base', () => {
    expect.assertions(1);
    const wrapper = render(<MatchDetailPanel />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('shouldComponentUpdate', () => {
    expect(3);
    const wrapper = mount(<MatchDetailPanel accordion={true} />);
    wrapper.setProps({
      activeKeyId: 'ff',
      activeStickKeyId: 'kk',
    });
    const showPanel = wrapper.state('showPanel');
    const isStick = wrapper.state('isStick');
    expect(false).toEqual(showPanel);
    expect(false).toEqual(isStick);
  });

  it('accordion-true', () => {
    expect.assertions(1);
    const wrapper = render(<MatchDetailPanel accordion={true} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('defaultHandlePanel', () => {
    expect.assertions(1);
    const wrapper = mount(<MatchDetailPanel accordion={true} />);
    const handlePanel = wrapper.prop('handlePanel');
    const click = jest.fn(handlePanel);
    wrapper
      .children('div')
      .at(0)
      .children('div')
      .at(0)
      .simulate('click');
    expect(click()).toBe('handlePanel');
  });
  it('defaultHandlePanelshowPanel-true', () => {
    expect.assertions(1);
    const wrapper = mount(<MatchDetailPanel accordion={true} />);
    wrapper.setState({
      showPanel: false,
    });
    const handlePanel = wrapper.prop('handlePanel');

    const click = jest.fn(handlePanel);
    wrapper
      .children('div')
      .at(0)
      .children('div')
      .at(0)
      .simulate('click');
    expect(click()).toBe('handlePanel');
  });
  it('defaultHandleStickIcon', () => {
    expect.assertions(1);
    const wrapper = mount(<MatchDetailPanel accordion={true} />);
    const handleStickIcon = wrapper.prop('handleStickIcon');
    const click = jest.fn(handleStickIcon);
    wrapper
      .children('div')
      .at(0)
      .children('div')
      .at(0)
      .children('div')
      .at(1)
      .children('span')
      .at(0)
      .simulate('click');
    expect(click()).toBe('handleStickIcon');
  });
  it('defaultHandleHelpIcon', () => {
    expect.assertions(1);
    const wrapper = mount(<MatchDetailPanel accordion={true} />);
    const handleHelpIcon = wrapper.prop('handleHelpIcon');
    const click = jest.fn(handleHelpIcon);
    wrapper
      .children('div')
      .at(0)
      .children('div')
      .at(0)
      .children('div')
      .at(0)
      .children('span')
      .at(0)
      .simulate('click');
    expect(click()).toBe('handleHelpIcon');
  });
});

describe('Panel', async () => {
  it('base', () => {
    expect.assertions(1);
    const wrapper = render(<Panel />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('prefixClsNUll', () => {
    expect.assertions(1);
    const wrapper = render(<Panel prefixCls={null} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('prefixClsNUll', () => {
    expect.assertions(1);
    const wrapper = render(<Panel accordion={true} activeKeyId="ff" keyId="ff" />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('defaultHandlePanel', () => {
    expect.assertions(1);
    const wrapper = mount(<Panel />);
    const handlePanel = wrapper.prop('handlePanel');
    const click = jest.fn(handlePanel);
    expect(click()).toBe('handlePanel');
  });
  it('handlePanel', () => {
    expect.assertions(1);
    const handlePanel = () => 'handlePanel';
    const click = jest.fn(handlePanel);
    const wrapper = mount(<Panel handlePanel={click} />);
    wrapper
      .children('div')
      .at(0)
      .children('div')
      .at(0)
      .simulate('click');
    expect(click()).toBe('handlePanel');
  });
  it('defaultHandlePanelShowPanel-false', () => {
    expect.assertions(1);
    const wrapper = mount(<Panel accordion={true} />);
    const handlePanel = wrapper.prop('handlePanel');
    const click = jest.fn(handlePanel);
    wrapper
      .children('div')
      .at(0)
      .children('div')
      .at(0)
      .simulate('click');
    expect(click()).toBe('handlePanel');
  });
  it('showPanel', () => {
    expect.assertions(1);
    const wrapper = mount(<Panel accordion={true} />);
    wrapper.setProps({
      activeKeyId: 'foodball',
    });
    const showPanel = wrapper.state('showPanel');
    expect(false).toEqual(showPanel);
  });
});
