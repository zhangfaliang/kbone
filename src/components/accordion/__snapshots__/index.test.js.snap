// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordion base 1`] = `
<div
  class="default-accordion"
/>
`;

exports[`Accordion base-accordion 1`] = `
<div
  class="default-accordion"
/>
`;

exports[`Accordion baseMatchDetailPanel 1`] = `
<div
  class="default-accordion"
>
  <div
    class="default-detail-panel"
  >
    <div
      class="title"
    >
      <div
        class="titleLife"
      >
        <span
          class="leftText leftTextP12"
        >
          Primera división de Liga
        </span>
      </div>
      <div
        class="titleRight"
      >
        <span
          class="stickIcon"
        >
          <span
            class="iconfont icon-zhiding"
          />
        </span>
        <span
          class="foldIcon"
        >
           
          <span
            class="iconfont icon-mycopy"
          />
        </span>
      </div>
    </div>
    <div
      class="children"
    />
  </div>
</div>
`;

exports[`Accordion basePanel 1`] = `
<div
  class="default-accordion"
>
  <div
    class="default-panel"
  >
    <div
      class="title"
    >
      <span
        class="text"
      >
        Primera división de Liga
      </span>
      <span
        class="iconFont"
      >
        <span
          class="iconfont icon-mycopy2"
        />
      </span>
    </div>
    <div
      class="children"
    />
  </div>
</div>
`;

exports[`MatchDetailPanel accordion-true 1`] = `
<div
  class="default-detail-panel show"
>
  <div
    class="title"
  >
    <div
      class="titleLife"
    >
      <span
        class="leftText leftTextP12"
      >
        Primera división de Liga
      </span>
    </div>
    <div
      class="titleRight"
    >
      <span
        class="stickIcon"
      >
        <span
          class="iconfont icon-zhiding"
        />
      </span>
      <span
        class="foldIcon activeFoldIcon"
      >
         
        <span
          class="iconfont icon-mycopy"
        />
      </span>
    </div>
  </div>
  <div
    class="children"
  />
</div>
`;

exports[`MatchDetailPanel base 1`] = `
<div
  class="default-detail-panel"
>
  <div
    class="title"
  >
    <div
      class="titleLife"
    >
      <span
        class="leftText leftTextP12"
      >
        Primera división de Liga
      </span>
    </div>
    <div
      class="titleRight"
    >
      <span
        class="stickIcon"
      >
        <span
          class="iconfont icon-zhiding"
        />
      </span>
      <span
        class="foldIcon"
      >
         
        <span
          class="iconfont icon-mycopy"
        />
      </span>
    </div>
  </div>
  <div
    class="children"
  />
</div>
`;

exports[`Panel base 1`] = `
<div
  class="default-panel"
>
  <div
    class="title"
  >
    <span
      class="text"
    >
      Primera división de Liga
    </span>
    <span
      class="iconFont"
    >
      <span
        class="iconfont icon-mycopy2"
      />
    </span>
  </div>
  <div
    class="children"
  />
</div>
`;

exports[`Panel prefixClsNUll 1`] = `
<div
  class="default-panel"
>
  <div
    class="title"
  >
    <span
      class="text"
    >
      Primera división de Liga
    </span>
    <span
      class="iconFont"
    >
      <span
        class="iconfont icon-mycopy2"
      />
    </span>
  </div>
  <div
    class="children"
  />
</div>
`;

exports[`Panel prefixClsNUll 2`] = `
<div
  class="default-panel show"
>
  <div
    class="title"
  >
    <span
      class="text"
    >
      Primera división de Liga
    </span>
    <span
      class="iconFont"
    >
      <span
        class="iconfont icon-mycopy2"
      />
    </span>
  </div>
  <div
    class="children"
  />
</div>
`;
